src/app/loadout/loadout.service.ts

```diff
@@ -179,10 +191,34 @@ function LoadoutService(): LoadoutServiceType {
     await saveLoadouts(reduxStore.getState().loadouts.loadouts);
   }
 
-  async function saveLoadout(loadout: Loadout): Promise<void> {
-    await getLoadouts(); // make sure we have loaded all loadouts first!
-    reduxStore.dispatch(actions.updateLoadout(loadout));
-    await saveLoadouts(reduxStore.getState().loadouts.loadouts);
+  async function saveLoadout(loadout: Loadout): Promise<Loadout | undefined> {
+    const loadouts = await getLoadouts();
+    const clashingLoadout = getClashingLoadout(loadouts, loadout);
+
+    if (!clashingLoadout) {
+      reduxStore.dispatch(actions.updateLoadout(loadout));
+      await saveLoadouts(reduxStore.getState().loadouts.loadouts);
+    }
+
+    return clashingLoadout;
+  }
+
+  function getClashingLoadout(loadouts: Loadout[], newLoadout: Loadout): Loadout | undefined {
+    return _.find(loadouts, (loadout) => {
```

Comment:
Use `loadouts.find` over the Lodash find.

Reply:
Done.